### PR TITLE
[`SwitchTransformers`] Fix return values

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
 
The previous version of the code would always return `None` values for the router losses, since the `if output_router_probs` was wrapper around `if labels is not None`. But router logits can still be computed wthout  labels. 
Also returns tensors instead of None, this follows our usual API, and is less prone to errors.

### CODE REVIEW

File: src/transformers/models/switch_transformers/modeling_switch_transformers.py [hunk 1]
```diff
@@ -1681,26 +1681,20 @@ def forward(
...
+            # Compute the router loss (z_loss + auxiliary loss) for each router in the encoder and decoder
+            encoder_router_logits, encoder_expert_indexes = self._unpack_router_logits(encoder_outputs[-1])
+            encoder_z_loss = router_z_loss_func(encoder_router_logits)
+            encoder_router_probs = nn.Softmax(dim=-1)(encoder_router_logits)
+            encoder_aux_loss = load_balancing_loss_func(encoder_router_probs, encoder_expert_indexes)
 
-            if output_router_logits:
-                # Compute the router loss (z_loss + auxiliary loss) for each router in the encoder and decoder
-                encoder_router_logits, encoder_expert_indexes = self._unpack_router_logits(
-                    encoder_outputs.router_probs
```
Comment (ArthurZucker) [inline src/transformers/models/switch_transformers/modeling_switch_transformers.py:1691]: `encoder_outputs.router_probs` cannot be accessed if `return_dict = False`

File: src/transformers/models/switch_transformers/modeling_switch_transformers.py [hunk 2]
```diff
@@ -1712,19 +1706,12 @@ def forward(
...
-                output += (
-                    encoder_z_loss,
-                    encoder_aux_loss,
-                    decoder_z_loss,
-                    decoder_aux_loss,
-                    *decoder_outputs[1:],
-                    *encoder_outputs,
-                )
-            else:
-                output += (*decoder_outputs[1:], *encoder_outputs)
```
Comment (ArthurZucker) [inline src/transformers/models/switch_transformers/modeling_switch_transformers.py:1725]: just reduce used lines

File: src/transformers/models/switch_transformers/modeling_switch_transformers.py [hunk 3]
```diff
@@ -1736,18 +1723,18 @@ def forward(
             decoder_hidden_states=decoder_outputs.hidden_states,
             decoder_attentions=decoder_outputs.attentions,
             cross_attentions=decoder_outputs.cross_attentions,
+            decoder_router_logits=decoder_outputs.router_probs,
```
Comment (ArthurZucker) [inline src/transformers/models/switch_transformers/modeling_switch_transformers.py:1726]: make order of return consistent with tuples

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!
